versatileimagefield/utils.py

```diff
@@ -163,14 +143,17 @@ def get_filtered_path(path_to_image, filename_key, storage):
     return path_to_return
 
 
-def get_image_metadata_from_file_ext(file_ext):
+def get_image_metadata_from_file(file_like):
     """
-    Receive a valid image file format and returns a 2-tuple of two strings:
+    Receive a valid image file and returns a 2-tuple of two strings:
         [0]: Image format (i.e. 'jpg', 'gif' or 'png')
         [1]: InMemoryUploadedFile-friendly save format (i.e. 'image/jpeg')
     image_format, in_memory_file_type
     """
-    return FILE_EXTENSION_MAP.get(file_ext, JPEG)
+    mime_type = magic.from_buffer(file_like.read(1024), mime=True)
+    file_like.seek(0)
+    image_format = MIME_TYPE_TO_PIL_IDENTIFIER[mime_type]
```

Comment:
What if the mime type is not an image?

Reply:
Thanks for the review. I was hesitating to address this issue. When the code was relying on the file extension, it defaulted to `JPEG`. Does that make sense? Should we keep that? I can name any file `.jpeg` and have it processed by this library. My guess would be, that it fails loudly. Then why should I default to JPEG only to fail loudly, later on, anyways? It seems to me, that this library is not responsible for doing validation. So that should be handled by the library user. Check the files for content types and don't let unsupported pass. Of course, we could do validation in this package, too. But that seems like a much bigger task / issue and not easy to fix.

Reply:
Or put another way: There seem to be no tests for unsupported / unknown image files. So nobody seems to care about that, right? ;-)

Cheers.

Reply:
Oh, and another note, that just came to my mind: We've had problems with defaulting to JPEG for some PNG files, which had the wrong extension. Somehow, the JPEG-engine was able to process the data, but images were just completely scrambled.